Backend/Tournament_User/TournamentUserMVSC/server.js

```diff
@@ -0,0 +1,181 @@
+const express = require('express');
+const app = express();
+const mysql = require('mysql2');
+
+// Set up the MySQL connection
+const db = mysql.createConnection({
+    host: '202.166.134.202',
+    user: 'root',   // Replace with your DB username
+    password: 'UnZ7Sqx26eFpwfBsLXC8KE',   // Replace with your DB password
+    port: 32781,
+    database: 'TournamentMSVC' // Replace with your DB name
+});
+
+// Connect to MySQL
+db.connect((err) => {
+    if (err) {
+        console.error('Error connecting to the database:', err);
+        return;
+    }
+    console.log('Connected to the MySQL database');
+});
+
+// Middleware to parse incoming JSON
+app.use(express.json());
+
+// Sample route to get tournament history
+app.get('/tournaments', (req, res) => {
+    const sql = `CALL GetTournamentHistory();`;
+    
+    db.query(sql, (err, results) => {
+        if (err) {
+            res.status(500).send('Error retrieving data from database');
+        } else {
+            res.status(200).json(results);
+        }
+    });
+});
+
+//Route to get tournamentId 
+
+app.get('/tournaments/GetTournamentMatchups/:tournamentId',(req,res)=>{
+    const tournamentId = req.params.tournamentId;
+    const getTournamentIdsql = 'CALL GetTournamentUsingID(?)';
+
+    db.query(getTournamentIdsql, [tournamentId], (err, results) => {
+        if (err) {
+            console.error('Error fetching matchups from the database:', err);
+            res.status(500).send('Failed to retrieve matchups');
+        } else if (results.length === 0) {
+            res.status(404).send('No matchups found for the provided tournament ID');
+        } else {
+            res.status(200).json(results);
+        }
+    });
+
+});
+
+// Middleware to parse incoming JSON
+app.use(express.json());
+
+
+
+
+
+// Route to sign up a user for a tournament using a stored procedure
+app.post('/tournaments/signup/:UUID', (req, res) => {
+    const UUID = req.params.UUID;   // UUID parsed from the URL
+    const { tournamentID, elo } = req.body;  // tournamentID and elo from the request body
+
+    // Check if required parameters are provided
+    if (!UUID || !tournamentID || !elo) {
+        return res.status(400).json({ error: 'Missing required fields (UUID, tournamentID, or elo)' });
+    }
+
+    // Call the stored procedure for signing up
+    const sql = 'CALL SignUpForTournament(?, ?, ?)';
+    
+    db.query(sql, [UUID, tournamentID, elo], (err, results) => {
```

Comment:
following the model-controller format, database interaction and routes should be separated, and typically the server configs itself is in another separate file (app/server/index.js), totalling up to about 3-4 files total, in similarly named folders.

so eg controller -> tournament_controller.js with package.json and server.js in the root of the application folder